impl/src/planning/subplanners.rs

```diff
@@ -412,32 +414,21 @@ impl<'planner> CrateSubplanner<'planner> {
     license::get_license_from_str(licenses_str)
   }
 
-  fn _produce_deps(&self, names: &DependencyNames) -> Result<DependencySet> {
-    let build_dep_names = &names.build_dep_names;
-    let dev_dep_names = &names.dev_dep_names;
-    let normal_dep_names = &names.normal_dep_names;
-    let aliased_dep_names = &names.aliased_dep_names;
-
-    let mut dep_set = DependencySet {
-      build_deps: Vec::new(),
-      build_proc_macro_deps: Vec::new(),
-      proc_macro_deps: Vec::new(),
-      dev_deps: Vec::new(),
-      normal_deps: Vec::new(),
-      aliased_deps: Vec::new(),
-    };
+  /** Generates the set of dependencies for the contained crate. */
+  fn produce_deps(&self) -> Result<HashMap<Option<&'planner str>, CrateDependencyContext>> {
+    let mut dep_production: HashMap<Option<&'planner str>, CrateDependencyContext> = HashMap::new();
 
     let all_skipped_deps = self
       .crate_settings
       .iter()
       .flat_map(|pkg| pkg.skipped_deps.iter())
       .collect::<HashSet<_>>();
 
-    for dep_id in &self.node.dependencies {
+    for dep in &self.node.deps {
```

Comment:
This wonderful part of the metadata gives us *both* _renames_ and _targets_. Irritatingly its still a little tricky to detect renames fully (we need this to finally deduce the aliases - see `is_renamed`)

If https://github.com/rust-lang/cargo/issues/7289 gets solved then a lot of the left-over rename detection code can get murdered